fe/fe-core/src/main/java/org/apache/doris/nereids/util/JoinUtils.java

```diff
@@ -311,6 +311,23 @@ public static List<Expression> replaceJoinConjuncts(List<Expression> joinConjunc
                 ).collect(ImmutableList.toImmutableList());
     }
 
+    /**
+     * replace JoinConjuncts by using Expression map.
+     */
+    public static List<Expression> replaceJoinConjunctsUseExpression(List<Expression> joinConjuncts,
+            Map<ExprId, Expression> replaceMaps) {
+        return joinConjuncts.stream()
+                .map(expr ->
+                        expr.rewriteUp(e -> {
+                            if (e instanceof Slot && replaceMaps.containsKey(((Slot) e).getExprId())) {
```

Comment:
It is better to use SlotReference as key of replaceMaps instead of exprId.
SlotReference.equals() compare Slot by its exprId, so there is no performance loose.